kvm: Properly report available memory to Management Server

wido opened the pull request:
## Description
The KVM Agent had two mechanisms for reporting its capabilities
and memory to the Management Server.

On startup it would ask libvirt the amount of Memory the Host has
and subtract and add the reserved and overcommit memory.

When the HostStats were however reported to the Management Server
these two configured values on the Agent were no longer reported
in the statistics thus showing all the available memory in the
Agent/Host to the Management Server.

This commit unifies this by using the same logic on Agent Startup
and during statistics reporting.

  memory=3069636608, reservedMemory=1073741824

This was reported by a 4GB Hypervisor with this setting:

  host.reserved.mem.mb=1024

The GUI (thus API) would then show:

  Memory Total	2.86 GB

This way the Agent properly 'lies' to the Management Server about its
capabilities in terms of Memory.

This is very helpful if you want to overprovision or undercommit machines
for various reasons.

Overcommitting can be done when KSM or ZSwap or a fast SWAP device is
installed in the machine.

Underprovisioning is done when the Host might run other tasks then a KVM
hypervisor, for example when it runs in a hyperconverged setup with Ceph.

In addition internally many values have been changed from a Double to a Long
and also store the amount of bytes instead of Kilobytes.

## Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

I tested this on a CloudStack 4.12 test setup running on my local system.

The KVM Agent will report the amount of memory it has minus any reservations. This way the Mgmt server will never know the exact amount of memory the Agent has.

## Checklist:
- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [x] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

dhlaluku commented:
@blueorangutan package

rhtyd commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/linux/HostInfo.java:
```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class HostInfo {
```
Can you rename this class to KVMHostInfo, that way it will be more obvious what this class is expected of?

rhtyd commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/linux/HostInfo.java:
```diff
@@ -0,0 +1,132 @@
+
+public class HostInfo {
+
+    private static final Logger s_logger = Logger.getLogger(HostInfo.class);
```
Remove the s_, instead using `LOG` or `LOGGER`?

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2237

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

wido commented:
@rhtyd Done, I've renamed the class and the logger variable. How does this look like?

wido commented:
Ping @RPDiep

blueorangutan commented:
<b>Trillian test result (tid-2925)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 31642 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t2925-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_primary_storage.py
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermitten failure detected: /marvin/tests/smoke/test_snapshots.py
Intermitten failure detected: /marvin/tests/smoke/test_templates.py
Intermitten failure detected: /marvin/tests/smoke/test_usage.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermitten failure detected: /marvin/tests/smoke/test_volumes.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermitten failure detected: /marvin/tests/smoke/test_host_maintenance.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 60 look OK, 9 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_add_primary_storage_disabled_host | `Error` | 0.57 | test_primary_storage.py
test_01_primary_storage_nfs | `Error` | 0.08 | test_primary_storage.py
ContextSuite context=TestStorageTags>:setup | `Error` | 0.15 | test_primary_storage.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 1104.41 | test_privategw_acl.py
test_02_list_snapshots_with_removed_data_store | `Error` | 1.11 | test_snapshots.py
test_04_extract_template | `Failure` | 128.24 | test_templates.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
test_01_secure_vm_migration | `Error` | 51.90 | test_vm_life_cycle.py
test_02_unsecure_vm_migration | `Error` | 122.96 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 14.66 | test_vm_life_cycle.py
test_06_download_detached_volume | `Failure` | 137.53 | test_volumes.py
test_01_cancel_host_maintenace_with_no_migration_jobs | `Failure` | 6.25 | test_host_maintenance.py
test_02_cancel_host_maintenace_with_migration_jobs | `Error` | 3.25 | test_host_maintenance.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 3.41 | test_hostha_kvm.py

dhlaluku commented:
@blueorangutan package

blueorangutan commented:
@dhlaluku a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2238

dhlaluku commented:
@blueorangutan test

blueorangutan commented:
@dhlaluku a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rafaelweingartner commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/linux/KVMHostInfo.java:
```diff
@@ -0,0 +1,132 @@
+            final LibvirtCapXMLParser parser = new LibvirtCapXMLParser();
+            parser.parseCapabilitiesXML(conn.getCapabilities());
+            final ArrayList<String> oss = parser.getGuestOsType();
+            for (final String s : oss) {
```
Why doing all of this process if you only report the host supporting HVM? I mean, KVM does not run if the CPU does not support HVM. Therefore, all KVM hosts will always have HVM support, right?

wido replied:
Yes, they will. But what I did is take the original code and use it. I didn't modify the logic behind the capabilities.

I would make that a different PR if we want to do that.

rafaelweingartner replied:
I understand that you are only moving the old code around, but why not remove useless code now?

I mean, why would we keep a code that does not aggregate value, and can be misleading to newcomers?

GabrielBrascher commented:
@rhtyd do you know if the failing tests are false positives? I have been testing some actions such as VM migrations and adding/removing hosts. I had no problem when using primary storage with nfs and a cluster with KVM servers.

dhlaluku commented:
@GabrielBrascher most of the failing tests have already been addressed and merged into master with this PRs #2763 #2756 #2754 

@wido have this changes been rebased against the latest master?

wido commented:
@dhlaluku I've rebased against master

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2256

dhlaluku commented:
@blueorangutan test

blueorangutan commented:
@dhlaluku a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2938)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 33709 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t2938-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 65 look OK, 4 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 1183.01 | test_privategw_acl.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 189.74 | test_vm_life_cycle.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 3644.93 | test_vm_life_cycle.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 3783.07 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 20.17 | test_vm_life_cycle.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 276.75 | test_vpc_redundant.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 3.83 | test_hostha_kvm.py

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✖centos7 ✖debian. JID-2305

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2318

DaanHoogland commented:
@blueorangutan test

blueorangutan commented:
@DaanHoogland a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3040)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 32310 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3040-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_primary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 62 look OK, 7 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 46.38 | test_multipleips_per_nic.py
test_01_add_primary_storage_disabled_host | `Error` | 0.80 | test_primary_storage.py
test_01_primary_storage_nfs | `Error` | 0.08 | test_primary_storage.py
ContextSuite context=TestStorageTags>:setup | `Error` | 0.16 | test_primary_storage.py
test_02_vpc_privategw_static_routes | `Failure` | 202.79 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 1239.70 | test_privategw_acl.py
test_02_list_snapshots_with_removed_data_store | `Error` | 1.11 | test_snapshots.py
test_01_secure_vm_migration | `Error` | 134.30 | test_vm_life_cycle.py
test_02_unsecure_vm_migration | `Error` | 133.78 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 18.65 | test_vm_life_cycle.py
test_01_cancel_host_maintenace_with_no_migration_jobs | `Failure` | 0.09 | test_host_maintenance.py
test_02_cancel_host_maintenace_with_migration_jobs | `Error` | 2.21 | test_host_maintenance.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 1.42 | test_hostha_kvm.py
test_hostha_kvm_host_fencing | `Failure` | 654.87 | test_hostha_kvm.py
test_hostha_kvm_host_recovering | `Failure` | 622.49 | test_hostha_kvm.py

wido commented:
The tests which were failing here do not seem related to the change. Is this one ready to merge?

dhlaluku commented:
@blueorangutan test

blueorangutan commented:
@dhlaluku a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3048)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 31071 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3048-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_templates.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 63 look OK, 6 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 23.03 | test_multipleips_per_nic.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 1095.31 | test_privategw_acl.py
test_04_extract_template | `Failure` | 128.32 | test_templates.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
test_06_download_detached_volume | `Failure` | 136.54 | test_volumes.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 3.43 | test_hostha_kvm.py

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✖debian. JID-2332

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rafaelweingartner commented on plugins/hypervisors/kvm/src/test/java/org/apache/cloudstack/utils/linux/KVMHostInfoTest.java:
```diff
@@ -0,0 +1,34 @@
+
+public class KVMHostInfoTest {
+    @Test
+    public void getCpuSpeed() {
```
A test case without an assert?

Also, why not mock things, and then run the test always? I mean, if somebody is using MacOS (not sure here) or Windows the test will be ignored.

rafaelweingartner commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/linux/MemStat.java:
```diff
@@ -22,30 +22,47 @@
 
+
 public class MemStat {
+    /*
```
Is this a documentation or a code comment?

rafaelweingartner commented on plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:
```diff
@@ -383,11 +378,11 @@ public LibvirtUtilitiesHelper getLibvirtUtilitiesHelper() {
 
     public CPUStat getCPUStat() {
-        return _cpuStat;
+        return new CPUStat();
```
Is this coded on purpose? I mean, can we use it as a stateless call? At first sight this `getCPUStat` seem to return an object property, but instead it will always return a different object instance.

blueorangutan commented:
<b>Trillian test result (tid-3059)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 24606 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3059-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 64 look OK, 5 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 25.09 | test_multipleips_per_nic.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 1123.63 | test_privategw_acl.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 103.06 | test_vm_life_cycle.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 457.60 | test_vpc_redundant.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 2.40 | test_hostha_kvm.py

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✖debian. JID-2379

wido commented:
It seems we are ready to merge this one?

rafaelweingartner commented:
I would say that I have unanswered questions...

wido commented:
I see @rafaelweingartner ! Missed those, I will look at them.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-2421

dhlaluku commented on plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:
```diff
@@ -383,11 +378,11 @@ public LibvirtUtilitiesHelper getLibvirtUtilitiesHelper() {
 
     public MemStat getMemStat() {
-        return _memStat;
+        return new MemStat(_dom0MinMem, _dom0OvercommitMem);
```
Same comment as above, the returned object instance is always different for every method call

rhtyd commented:
@wido this is failing build, can you check?

wido commented:
I will @rhtyd 

Currently in the middle of VXLAN testing, so postponing this one for a moment. Need to pick it up soon and go through the comments.

wido commented:
@rafaelweingartner I think I addressed your comments.

I left the 'hvm' code in there as it does serve a purpose. If the node the Agent is started on does not have VT enabled in the BIOS we throw an error and also report to the mgmt server we are not HVM capable. So that's good.

Otherwise I also fixed the Unit Test, so I think we are good :)

rafaelweingartner commented:
forcing Jenkins

rafaelweingartner commented:
@wido Jenkins is not being able to build this PR. Can this be a dependency problem?
```
[ERROR] Errors: 
[ERROR]   KVMHostInfoTest.getCpu:39 » UnsatisfiedLink Unable to load library 'virt': Nat...
[ERROR]   KVMHostInfoTest.getMemory:47 » NoClassDefFound Could not initialize class org....
```

wido commented:
Jenkins is happy now. How are we with this PR? Can we merge it?

All comments seem to be adressed

rhtyd commented:
@wido let's test one last time
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-2517

rhtyd commented:
@wido can you check and fix build failures, see job 1 of travis

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-2518

wido commented:
I just force pushed. This seems to be a thing (Unit Test) which fails on Travis, but works on my local system.

Could be a problem where tests are running inside a Container?

GabrielBrascher commented:
@blueorangutan package

blueorangutan commented:
@GabrielBrascher a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2551

GabrielBrascher commented:
@blueorangutan test

blueorangutan commented:
@GabrielBrascher a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rafaelweingartner commented:
@wido we need to check the integration test results before being able to merge it.
Thanks for the PR.

blueorangutan commented:
<b>Trillian test result (tid-3333)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 26958 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3333-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_primary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 66 look OK, 4 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_add_primary_storage_disabled_host | `Error` | 0.87 | test_primary_storage.py
test_01_primary_storage_nfs | `Error` | 0.15 | test_primary_storage.py
ContextSuite context=TestStorageTags>:setup | `Error` | 0.28 | test_primary_storage.py
test_02_list_snapshots_with_removed_data_store | `Error` | 1.17 | test_snapshots.py
test_11_migrate_vm | `Error` | 79.50 | test_vm_life_cycle.py
test_14_secure_to_secure_vm_migration | `Error` | 56.54 | test_vm_life_cycle.py
test_01_cancel_host_maintenace_with_no_migration_jobs | `Failure` | 0.13 | test_host_maintenance.py
test_02_cancel_host_maintenace_with_migration_jobs | `Error` | 2.34 | test_host_maintenance.py

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3335)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 21178 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3335-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_primary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Smoke tests completed. 66 look OK, 4 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_add_primary_storage_disabled_host | `Error` | 0.71 | test_primary_storage.py
test_01_primary_storage_nfs | `Error` | 0.12 | test_primary_storage.py
ContextSuite context=TestStorageTags>:setup | `Error` | 0.24 | test_primary_storage.py
test_02_list_snapshots_with_removed_data_store | `Error` | 1.16 | test_snapshots.py
test_11_migrate_vm | `Error` | 16.91 | test_vm_life_cycle.py
test_14_secure_to_secure_vm_migration | `Error` | 70.87 | test_vm_life_cycle.py
test_01_cancel_host_maintenace_with_no_migration_jobs | `Failure` | 0.14 | test_host_maintenance.py
test_02_cancel_host_maintenace_with_migration_jobs | `Error` | 2.39 | test_host_maintenance.py

GabrielBrascher commented:
@blueorangutan package

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2555

GabrielBrascher commented:
@blueorangutan test

blueorangutan commented:
@GabrielBrascher a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3338)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 21808 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2795-t3338-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Smoke tests completed. 69 look OK, 1 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_redundant_vpc_site2site_vpn | `Failure` | 206.83 | test_vpc_vpn.py
